### ISSUE TITLE
How to use query scope with hasManyDeepFromRelations?

### ISSUE THREAD

**Issue by Siddharaj6010** [staudenmeir/eloquent-has-many-deep#36]
Thank you for this wonderful package. I just came to know about it recently and it has already made many parts of my code simpler. 

I went through the documentation but there's one problem where I would appreciate some help. 

I have the following relationship:

- **Lawyer office** hasMany **Lawyers**
- **Lawyer** belongsToMany **Companies**
- **Company** belongsToMany **Lawyers**
- **Company** hasMany **Claims**

Now I was able to get the companies of the lawyer office by using **hasManyDeepFromRelations** as 
```
// Inside LawyerOffice Model

public function companies()
{
    return $this->hasManyDeepFromRelations($this->lawyers(), (new Lawyer)->companies())->active();
}
```

But when I do the same for claims as 
```
public function claims()
{
    return $this->hasManyDeepFromRelations($this->companies(), (new Company)->claims());
}
```
It ignores the **active()** query scope of the companies and returns claims of all the companies. 

I assume this is something related to #35 but I am not sure where should I add the query scope manually. 

Can you please guide me on how can I add query scope or **where()** condition in order to get the claims of the active companies of a lawyer office. 

Thank you.

**Comment by staudenmeir** [comment]
It's not explicitly stated in the documentation, but `hasManyDeepFromRelations()` doesn't apply additional constraints from the original relationships.

You'll need to add the scope's constraint(s) manually:

```php
public function claims()
{
    return $this->hasManyDeepFromRelations($this->companies(), (new Company)->claims())
        ->where('companies.active', true);
}
```

**Comment by Siddharaj6010** [comment]
This is working exceptionally well. 

Thank you for the quick response :)

**Comment by hasnatbabur** [comment]
@staudenmeir Thanks for this awesome package.

For this situation:
"It's not explicitly stated in the documentation, but hasManyDeepFromRelations() doesn't apply additional constraints from the original relationships."

Don't you think this is violating the DRY rule? Let's say I want to get all posts which is only published most cases. So if I forgot to add the publish constraint then different query will behave different and there is no single source of control on a relationship. A lots of bug will be left under the hood.

Do you think we can improve the issue? Thanks

**Comment by staudenmeir** [comment]
@hasnatbabur I agree, it wasn't a deliberate decision to _not_ apply the constraints. The issue is quite complex and I'm still working on it.

I see the automatic concatenation of foreign and local keys as the main benefit of `hasManyDeepFromRelation()` and decided to release it without the constraint "feature".

**Comment by staudenmeir** [comment]
I released `v1.15.3` with [`hasManyDeepFromRelationsWithConstraints()`](https://github.com/staudenmeir/eloquent-has-many-deep#existing-relationships-constraints):

```php
public function claims()
{
    return $this->hasManyDeepFromRelationsWithConstraints([$this, 'companies'], [new Company(), 'claims']);
}
```
